### PR TITLE
Fix integration tests TestRecoverFromInvalidOutputConfiguration

### PR DESCRIPTION
**Pull request by belimawr**
:warning: **Auto-merge is enabled** :warning:

## Proposed commit message

TestRecoverFromInvalidOutputConfiguration was failing due to a closed channel being closed again. Since 0e9c9de was introduced the Filestream input is reporting its status, which leads Filebeat to make more CheckIn calls to the Elastic-Agent. The test could not handle the extra calls.

A boolean atomic flag has been introduced to indicate the channel has already been closed.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Disruptive User Impact

None, this PR only fixes a broken test

~~## Author's Checklist~~
## How to test this PR locally
Run the test that was failing: 
```
cd x-pack/filebeat
mage clean buildSystemTestBinary
go test -tags=integration ./tests/integration -count=1 -run=TestRecoverFromInvalidOutputConfiguration
```

~~## Related issues~~
~~## Use cases~~
~~## Screenshots~~
~~## Logs~~

### CODE REVIEW

**Review by jlind23 — APPROVED** [review]

File: x-pack/filebeat/tests/integration/managerV2_test.go [hunk 1]
```diff
@@ -504,9 +504,17 @@ func TestRecoverFromInvalidOutputConfiguration(t *testing.T) {
...
+	// succeededChannelClosed is set to true whenever the `success`
+	// channel is closed. The Filestream input now is reporting its state
+	// to the Elastic-Agent, which cases more checkins to happen, thus the
+	// `success` channel was being close twice. `succeededChannelClosed`
+	// prevents that from happening.
+	succeededChannelClosed := atomic.Bool{}
 	// The test is successful when we reach the last element of `protoUnits`
 	onObserved := func(observed *proto.CheckinObserved, protoUnitsIdx int) {
-		if protoUnitsIdx == len(protos)-1 {
+		if protoUnitsIdx == len(protos)-1 && !succeededChannelClosed.Load() {
```
Comment (leehinman) [inline x-pack/filebeat/tests/integration/managerV2_test.go:516]: I'm not sure this guarantees that close will only get called once.  Is it possible for the condition to be evaluated, then go routine put to sleep and another go routine run the same code and call close?  In that case we would still call close twice.

Could we just use the `sync.Once` pattern to make sure that close is only called once?

Comment (belimawr) [reply x-pack/filebeat/tests/integration/managerV2_test.go:516]: You're right Lee, I'll make this improvement PR.

Comment (belimawr) [reply x-pack/filebeat/tests/integration/managerV2_test.go:516]: Here is the PR with the fix: https://github.com/elastic/beats/pull/40199

**Comment by belimawr** [comment]
> I should have updated this in my first PR itself. Sorry that I missed it, my bad 🙏

No worries, we need to get those tests running on CI.

**Pull request merged**